Grocery_List_Zhecan_Wang/app/src/main/java/com/example/jong/todolist/MainActivity.java

```diff
@@ -0,0 +1,129 @@
+package com.example.jong.todolist;
+
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.ListView;
+
+import java.util.ArrayList;
+
+public class MainActivity extends AppCompatActivity {
+    
+    ArrayList<String> arraylist = new ArrayList();
+    ListView listview;
+               
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        listview = (ListView) findViewById(R.id.todolist1);
+        listview.setAdapter(new ArrayAdapter<String>(this, R.layout.task_view, arraylist));
+        listview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                final int pos = position;
+                String entry = arraylist.get(position);
+                AlertDialog.Builder edittaskalert = new AlertDialog.Builder(MainActivity.this);
+                edittaskalert.setTitle("Edit Task");
+                edittaskalert.setMessage("Need to do:");
+                final EditText textinput = new EditText(MainActivity.this);
+                textinput.setText(entry);
+                edittaskalert.setView(textinput);
+                DialogInterface.OnClickListener Editbuttonlistener = new DialogInterface.OnClickListener() {
+
+                    @Override
+                    public void onClick(DialogInterface dialogInterface, int i) {
+                        String task = textinput.getText().toString();
+                        Log.d("MainActivity", task);
+                        arraylist.set(pos, task);
+                        ArrayAdapter<String> todolist_adapter = (ArrayAdapter<String>) listview.getAdapter();
+                        todolist_adapter.notifyDataSetChanged();
+                    }
+                };
+                edittaskalert.setPositiveButton("Edit", Editbuttonlistener);
+
+                edittaskalert.setNegativeButton("Cancel", null);
+                edittaskalert.create().show();
+
+
+            }
+        });
+
+        listview.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
```

Comment:
You could also define an implementation of this in a separate class to keep this file cleaner.